search/search_repository.go

```diff
@@ -387,7 +394,11 @@ func (q Query) generateExpression() (criteria.Expression, error) {
 			if q.Negate {
 				myexpr = append(myexpr, criteria.Not(left, right))
 			} else {
-				myexpr = append(myexpr, criteria.Equals(left, right))
+				if q.Substring {
```

Comment:
This looks like a hack and not very modular but we can address that with the next modifier coming in.